File: libs/sdk/android/src/main/java/com/rudderstack/react/android/RNRudderAnalytics.java
package com.rudderstack.react.android;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.rudderstack.android.sdk.core.RudderConfig;
import com.rudderstack.android.sdk.core.RudderIntegration;
import com.rudderstack.android.sdk.core.consent.RudderConsentFilter;

import java.util.ArrayList;
import java.util.List;

public class RNRudderAnalytics {
    static List<RudderIntegration.Factory> integrationList;
    static RudderConfig.DBEncryption rsDBEncryption = null;
    private static RudderConsentFilter rsConsentFilter;

    public static void addIntegration(RudderIntegration.Factory integration) {
        if (integrationList == null) {
            integrationList = new ArrayList<>();
        }
        integrationList.add(integration);
    }

    public static RudderConfig buildWithIntegrations(RudderConfig.Builder builder) {
        if (integrationList != null) {
            builder.withFactories(integrationList);
        }
        return builder.build();
    }

    public static void setDBEncryption(RudderConfig.DBEncryption dbEncryption) {
        rsDBEncryption = dbEncryption;
    }

    @Nullable
    public static RudderConfig.DBEncryption getDBEncryption() {
        return rsDBEncryption;
    }

    public static void setConsentFilterPlugin(@NonNull RudderConsentFilter consentFilter) {
        rsConsentFilter = consentFilter;
    }

    @Nullable
    public static RudderConsentFilter getConsentFilterPlugin() {
        return rsConsentFilter;
    }
}

File: libs/sdk/ios/RNRudderAnalytics.m
//
//  RNRudderAnalytics.m
//  CocoaAsyncSocket
//
//  Created by Dhawal on 28/07/20.
//

#import "RNRudderAnalytics.h"

NSMutableArray* integrationList;
static RSDBEncryption* rsDBEncryption = nil;
static id<RSConsentFilter> rsConsentFilter = nil;

@implementation RNRudderAnalytics


+ (void) addIntegration:(id<RSIntegrationFactory>)integration {
    if (integrationList == nil) {
        integrationList = [[NSMutableArray alloc] init];
    }
    [integrationList addObject:integration];
}

+ (RSConfig*) buildWithIntegrations:(RSConfigBuilder*)builder {
    if (integrationList != nil) {
        for (id<RSIntegrationFactory> integration in integrationList) {
            [builder withFactory:integration];
        }
    }
    return [builder build];
}

+ (void) setDBEncryption:(RSDBEncryption *)dbEncryption {
    rsDBEncryption = dbEncryption;
}

+ (RSDBEncryption *_Nullable) getDBEncryption {
    return rsDBEncryption;
}

+ (void) setConsentFilterPlugin:(id <RSConsentFilter> _Nonnull)consentFilter {
    rsConsentFilter = consentFilter;
}

+ (id <RSConsentFilter> _Nullable) getConsentFilterPlugin {
    return rsConsentFilter;
}

@end
